src/dune/foreign.ml

```diff
@@ -102,6 +102,69 @@ let possible_sources ~language obj ~dune_version =
         (Language.equal lang language && dune_version >= version)
         (obj ^ "." ^ ext))
 
+module Archive = struct
+  module Name = struct
+    module T = struct
+      type t = string
+
+      let compare = String.compare
+
+      let to_dyn = String.to_dyn
+    end
+
+    include T
+    include Comparable.Make (T)
```

Comment:
You might as well just `include String` here

Reply:
Indeed, done in the upcoming commit.